jetson/ra_kinematics/mrover_arm.cpp

```diff
@@ -599,24 +577,43 @@ void MRoverArm::encoder_angles_sender() {
 }
 
 void MRoverArm::publish_config(const std::vector<double> &config, std::string channel) {
-       RAPosition ra_position;
-       ra_position.joint_a = config[0];
-       ra_position.joint_b = config[1];
-       ra_position.joint_c = config[2];
-       ra_position.joint_d = config[3];
-       ra_position.joint_e = config[4];
-       ra_position.joint_f = config[5];
-       lcm_.publish(channel, &ra_position); //no matching call to publish should take in const msg type msg
+    RAPosition ra_position;
+    
+    if (arm_state.num_joints() == 6) {
+        ra_position.joint_a = config[0];
+        ra_position.joint_b = config[1];
+        ra_position.joint_c = config[2];
+        ra_position.joint_d = config[3];
+        ra_position.joint_e = config[4];
+        ra_position.joint_f = config[5];
+    }
+    else {
+        ra_position.joint_a = config[0];
+        ra_position.joint_b = config[1];
+        ra_position.joint_c = config[2];
+        ra_position.joint_d = config[3];
```

Comment:
should be joint a b c e

Reply:
I suggest making this a virtual function and change it so that the string name of the channel is defined inside the publish_config function (since for the science arm, it should be publishing to /sa_position and using the SAPosition.lcm while for the standard arm, it should use /ra_position and use the RAPosition.lcm).

Reply:
We can't define the channel name inside `publish_config()` because in `MRoverArm::execute_spline()` there's this call: `publish_config(target_angles, "/ra_ik_cmd")`, so unless this line is incorrect and needs to be changed, we have to keep the channel name as its own parameter.

Reply:
The SA version of the channel "/ra_ik_cmd" was previously "/sa_closedloop_cmd" but it will be "sa_ik_cmd"